tests/api2/test_alert.py

```diff
@@ -0,0 +1,82 @@
+from time import sleep
+
+import pytest
+from pytest_dependency import depends
+
+from auto_config import pool_name
+from middlewared.test.integration.utils import call, ssh
+
+
+ID_PATH = "/dev/disk/by-partuuid/"
+
+
+def get_alert_by_id(alert_id):
+    return next(filter(lambda alert: alert["id"] == alert_id, call("alert.list")), None)
+
+
+@pytest.mark.dependency(name="degrade_pool")
+def test_degrading_a_pool_to_create_an_alert(request):
+    get_pool = call("pool.query", [["name", "=", pool_name]], {"get": True})
+    gptid = get_pool["topology"]["data"][0]["path"].replace(ID_PATH, "")
+    ssh(f"zinject -d {gptid} -A fault {pool_name}")
+    request.config.cache.set("alert/gptid", gptid)
+
+
+def test_verify_the_pool_is_degraded(request):
+    depends(request, ["degrade_pool"], scope="session")
+    gptid = request.config.cache.get("alert/gptid", "Not a valid id")
+    status = call("zpool.status", {"name": pool_name})[pool_name][ID_PATH + gptid]["disk_status"]
+    assert status == "DEGRADED"
+
+
+@pytest.mark.timeout(120)
+@pytest.mark.dependency(name="set_alert_id")
+def test_wait_for_the_alert_and_get_the_id(request):
+    depends(request, ["degrade_pool"], scope="session")
+    call("alert.process_alerts")
+    while True:
```

Comment:
You can just run `alert.run_source` to test what specific alert source returns. Please see other tests for examples.

Reply:
Will change

Reply:
Had to undo this change because `alert.run_source` does not return `id`.